### ISSUE TITLE
[BUG] mlflow.tensorflow.autolog() does not work with tensorflow object detection api.

### ISSUE THREAD

**Issue by zishanahmed08** [mlflow/mlflow#1964]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 18.04
- **MLflow installed from (source or binary)**:binary
- **MLflow version (run ``mlflow --version``)**:1.3.0
- **Python version**:3.6.5
- **npm version, if running the dev UI**:

### Describe the problem
I am trying to use the **mlflow.tensorflow.autolog()** with [tensorflow object detection api](https://github.com/tensorflow/models/tree/master/research/object_detection). Adding **mlflow.tensorflow.autolog()** to **[model_main.py](https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py)** logs some parameters like global_norm/clipped_gradient_norm,global_norm/gradient_norm,global_step/sec,learning_rate_1,loss_1,loss_2 in mlflow. However the more important metrics linke map,precision,recall are not being logged in mlfow even if they are present in tensorboard.I used the example [here](https://github.com/mlflow/mlflow/blob/master/examples/tensorflow/train_predict.py) as reference but did

**Comment by manesioz** [comment]
Please try to use descriptive issue titles, makes it easier for everyone to navigate :)

**Comment by zishanahmed08** [comment]
`
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=26.16s).
Accumulating evaluation results...
DONE (t=1.77s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.055
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.135
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.037
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.008
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.039
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.229
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.036
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.094
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.119
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.017
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.127
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.379
INFO:tensorflow:Finished evaluation at 2019-11-11-23:30:31
I1112 00:30:31.734762 140145783822144 evaluation.py:277] Finished evaluation at 2019-11-11-23:30:31
INFO:tensorflow:Saving dict for global step 100000: DetectionBoxes_Precision/mAP = 0.054754764, DetectionBoxes_Precision/mAP (large) = 0.22850765, DetectionBoxes_Precision/mAP (medium) = 0.038879838, DetectionBoxes_Precision/mAP (small) = 0.00770795, DetectionBoxes_Precision/mAP@.50IOU = 0.13532944, DetectionBoxes_Precision/mAP@.75IOU = 0.036984887, DetectionBoxes_Recall/AR@1 = 0.035873294, DetectionBoxes_Recall/AR@10 = 0.09437874, DetectionBoxes_Recall/AR@100 = 0.11854222, DetectionBoxes_Recall/AR@100 (large) = 0.3789076, DetectionBoxes_Recall/AR@100 (medium) = 0.12682755, DetectionBoxes_Recall/AR@100 (small) = 0.01683864, Loss/BoxClassifierLoss/classification_loss = 0.08574966, Loss/BoxClassifierLoss/localization_loss = 0.06854623, Loss/RPNLoss/localization_loss = 0.32471403, Loss/RPNLoss/objectness_loss = 0.12054889, Loss/total_loss = 0.59955907, global_step = 100000, learning_rate = 0.0002, loss = 0.59955907
I1112 00:30:31.734951 140145783822144 estimator.py:1979] Saving dict for global step 100000: DetectionBoxes_Precision/mAP = 0.054754764, DetectionBoxes_Precision/mAP (large) = 0.22850765, DetectionBoxes_Precision/mAP (medium) = 0.038879838, DetectionBoxes_Precision/mAP (small) = 0.00770795, DetectionBoxes_Precision/mAP@.50IOU = 0.13532944, DetectionBoxes_Precision/mAP@.75IOU = 0.036984887, DetectionBoxes_Recall/AR@1 = 0.035873294, DetectionBoxes_Recall/AR@10 = 0.09437874, DetectionBoxes_Recall/AR@100 = 0.11854222, DetectionBoxes_Recall/AR@100 (large) = 0.3789076, DetectionBoxes_Recall/AR@100 (medium) = 0.12682755, DetectionBoxes_Recall/AR@100 (small) = 0.01683864, Loss/BoxClassifierLoss/classification_loss = 0.08574966, Loss/BoxClassifierLoss/localization_loss = 0.06854623, Loss/RPNLoss/localization_loss = 0.32471403, Loss/RPNLoss/objectness_loss = 0.12054889, Loss/total_loss = 0.59955907, global_step = 100000, learning_rate = 0.0002, loss = 0.59955907
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 100000: /home/zahmed/mapillary/model/1/model.ckpt-100000
I1112 00:30:31.915772 140145783822144 estimator.py:2039] Saving 'checkpoint_path' summary for global step 100000: /home/zahmed/mapillary/model/1/model.ckpt-100000
INFO:tensorflow:Performing the final export in the end of training.
I1112 00:30:31.916367 140145783822144 exporter.py:415] Performing the final export in the end of training.
INFO:tensorflow:Calling model_fn.
I1112 00:30:32.054227 140145783822144 estimator.py:1111] Calling model_fn.
INFO:tensorflow:Scale of 0 disables regularizer.
I1112 00:30:33.098529 140145783822144 regularizers.py:98] Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
I1112 00:30:33.108535 140145783822144 regularizers.py:98] Scale of 0 disables regularizer.
INFO:tensorflow:depth of additional conv before box predictor: 0
I1112 00:30:33.108697 140145783822144 convolutional_box_predictor.py:148] depth of additional conv before box predictor: 0
INFO:tensorflow:Scale of 0 disables regularizer.
I1112 00:30:33.849303 140145783822144 regularizers.py:98] Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
I1112 00:30:33.861013 140145783822144 regularizers.py:98] Scale of 0 disables regularizer.
INFO:tensorflow:Done calling model_fn.
I1112 00:30:34.366137 140145783822144 estimator.py:1113] Done calling model_fn.
WARNING:tensorflow:From /home/zahmed/virtual_envs/tf111/lib/python3.6/site-packages/tensorflow/python/saved_model/signature_def_utils_impl.py:205: build_tensor_info (from tensorflow.python.saved_model.utils_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.utils.build_tensor_info or tf.compat.v1.saved_model.build_tensor_info.
W1112 00:30:34.366312 140145783822144 deprecation.py:323] From /home/zahmed/virtual_envs/tf111/lib/python3.6/site-packages/tensorflow/python/saved_model/signature_def_utils_impl.py:205: build_tensor_info (from tensorflow.python.saved_model.utils_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.utils.build_tensor_info or tf.compat.v1.saved_model.build_tensor_info.
INFO:tensorflow:Signatures INCLUDED in export for Classify: None
I1112 00:30:34.366615 140145783822144 export.py:587] Signatures INCLUDED in export for Classify: None
INFO:tensorflow:Signatures INCLUDED in export for Regress: None
I1112 00:30:34.366685 140145783822144 export.py:587] Signatures INCLUDED in export for Regress: None
INFO:tensorflow:Signatures INCLUDED in export for Predict: ['tensorflow/serving/predict', 'serving_default']
I1112 00:30:34.366729 140145783822144 export.py:587] Signatures INCLUDED in export for Predict: ['tensorflow/serving/predict', 'serving_default']
INFO:tensorflow:Signatures INCLUDED in export for Train: None
I1112 00:30:34.366768 140145783822144 export.py:587] Signatures INCLUDED in export for Train: None
INFO:tensorflow:Signatures INCLUDED in export for Eval: None
I1112 00:30:34.366806 140145783822144 export.py:587] Signatures INCLUDED in export for Eval: None
2019-11-12 00:30:34.367013: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-11-12 00:30:34.367039: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-11-12 00:30:34.367044: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2019-11-12 00:30:34.367048: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2019-11-12 00:30:34.367099: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 4671 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1060 6GB, pci bus id: 0000:01:00.0, compute capability: 6.1)
INFO:tensorflow:Restoring parameters from /home/zahmed/mapillary/model/1/model.ckpt-100000
I1112 00:30:34.576051 140145783822144 saver.py:1270] Restoring parameters from /home/zahmed/mapillary/model/1/model.ckpt-100000
INFO:tensorflow:Assets added to graph.
I1112 00:30:34.863874 140145783822144 builder_impl.py:654] Assets added to graph.
INFO:tensorflow:No assets to write.
I1112 00:30:34.863988 140145783822144 builder_impl.py:449] No assets to write.
INFO:tensorflow:SavedModel written to: /home/zahmed/mapillary/model/1/export/Servo_0/temp-b'1573515031'/saved_model.pb
I1112 00:30:35.407828 140145783822144 builder_impl.py:414] SavedModel written to: /home/zahmed/mapillary/model/1/export/Servo_0/temp-b'1573515031'/saved_model.pb
2019/11/12 00:30:35 INFO mlflow.tensorflow: Validating the specified TensorFlow model by attempting to load it in a new TensorFlow graph...
2019-11-12 00:30:35.424838: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2019-11-12 00:30:35.424865: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-11-12 00:30:35.424870: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2019-11-12 00:30:35.424874: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2019-11-12 00:30:35.424925: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 4671 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1060 6GB, pci bus id: 0000:01:00.0, compute capability: 6.1)
WARNING:tensorflow:From /home/zahmed/virtual_envs/tf111/lib/python3.6/site-packages/mlflow/tensorflow.py:310: load (from tensorflow.python.saved_model.loader_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.loader.load or tf.compat.v1.saved_model.load. There will be a new function for importing SavedModels in Tensorflow 2.0.
W1112 00:30:35.425210 140145783822144 deprecation.py:323] From /home/zahmed/virtual_envs/tf111/lib/python3.6/site-packages/mlflow/tensorflow.py:310: load (from tensorflow.python.saved_model.loader_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.loader.load or tf.compat.v1.saved_model.load. There will be a new function for importing SavedModels in Tensorflow 2.0.
INFO:tensorflow:Restoring parameters from /home/zahmed/mapillary/model/1/export/Servo_0/1573515031/variables/variables
I1112 00:30:36.170154 140145783822144 saver.py:1270] Restoring parameters from /home/zahmed/mapillary/model/1/export/Servo_0/1573515031/variables/variables
/home/zahmed/virtual_envs/tf111/lib/python3.6/site-packages/mlflow/tensorflow.py:615: UserWarning: Logging to MLflow failed: Could not find signature def key predict. Available keys are: ['tensorflow/serving/predict', 'serving_default']
  artifact_path='model')
INFO:tensorflow:Loss for final step: 0.4581113.
I1112 00:30:36.737250 140145783822144 estimator.py:359] Loss for final step: 0.4581113.`

**Comment by zishanahmed08** [comment]
A log of the user warnng is also attached in the above comment.

> UserWarning: Logging to MLflow failed: Could not find signature def key predict. Available keys are: ['tensorflow/serving/predict', 'serving_default'] artifact_path='model')

@manesioz Please let me know if you need anything else

**Comment by macimovic** [comment]
I have the same problem, can we get some help, please?

**Comment by Zeleni9** [comment]
I am also looking for solution to log all stdout and tensorflow.logging.INFO data to the file via ML Flow, but there are not any working examples for this use-case. Trying redirecting output of program to text file does not give Tensorflow.logging.INFO and mlflow.tensorflow.autolog() does not give all the info. Thank you.

**Comment by gloin01** [comment]
I know this issue is quite old, but I don't see any mentions to the workaround. So I just want to hint at issue #3367 
And here a direct link to a partial solution:
https://github.com/mlflow/mlflow/issues/3367#issuecomment-730594616

**Comment by chenmoneygithub** [comment]
I believe the right solution is to write a custom MLflow callback, similar to [`mlflow.tensorflow.MLflowCallback`](https://github.com/mlflow/mlflow/blob/2d105c0896f3d2c3a9d96618438787e91487a2a8/mlflow/tensorflow/callback.py#L7). Since this is an old issue I am closing it, we shall reopen it if this issue bumps back.
